app/init/global_event_handler.js

```diff
@@ -249,9 +250,10 @@ class GlobalEventHandler {
     onServerVersionChanged = async (serverVersion) => {
         const {dispatch, getState} = Store.redux;
         const state = getState();
-        if (serverVersion && state.entities.users && state.entities.users.currentUserId) {
+        const isValid = validVersion(serverVersion);
+        if (isValid && serverVersion && serverVersion !== state.entities.general.serverVersion &&
```

Comment:
Personal preference: There seems to be too many conditions in this if. For readability sake, should we separate each AND clause in a different line?